crates/factor-outbound-redis/src/factor_redis.rs

```diff
@@ -0,0 +1,304 @@
+use anyhow::Result;
+use redis::{aio::Connection, AsyncCommands, FromRedisValue, Value};
+use spin_core::{async_trait, wasmtime::component::Resource};
+use spin_world::v1::{redis as v1, redis_types};
+use spin_world::v2::redis::{
+    self as v2, Connection as RedisConnection, Error, RedisParameter, RedisResult,
+};
+
+use tracing::{instrument, Level};
+
+struct RedisResults(Vec<RedisResult>);
+
+impl FromRedisValue for RedisResults {
```

Comment:
Nit: can we move this helper stuff to the bottom of the file. I think we typically have the more import things at the top of the file.